src/Core/Checkout/Payment/PaymentMethod/PaymentMethodExtension.php

```diff
@@ -0,0 +1,51 @@
+<?php declare(strict_types=1);
+/**
+ *                       ######
+ *                       ######
+ * ############    ####( ######  #####. ######  ############   ############
+ * #############  #####( ######  #####. ######  #############  #############
+ *        ######  #####( ######  #####. ######  #####  ######  #####  ######
+ * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
+ * ###### ######  #####( ######  #####. ######  #####          #####  ######
+ * #############  #############  #############  #############  #####  ######
+ *  ############   ############  #############   ############  #####  ######
+ *                                      ######
+ *                               #############
+ *                               ############
+ *
+ * Adyen Payment Module
+ *
+ * Copyright (c) 2021 Adyen B.V.
+ * This file is open source and available under the MIT license.
+ * See the LICENSE file for more info.
+ *
+ * Author: Adyen <shopware@adyen.com>
+ */
+
+namespace Adyen\Shopware\Core\Checkout\Payment\PaymentMethod;
+
+use Shopware\Core\Checkout\Payment\PaymentMethodDefinition;
+use Shopware\Core\Framework\DataAbstractionLayer\EntityExtension;
+use Shopware\Core\Framework\DataAbstractionLayer\Field\Flag\ApiAware;
+use Shopware\Core\Framework\DataAbstractionLayer\Field\Flag\Computed;
+use Shopware\Core\Framework\DataAbstractionLayer\Field\Flag\Runtime;
+use Shopware\Core\Framework\DataAbstractionLayer\Field\ObjectField;
+use Shopware\Core\Framework\DataAbstractionLayer\FieldCollection;
+
+class PaymentMethodExtension extends EntityExtension
+{
+    public function extendFields(FieldCollection $collection): void
+    {
+        $collection->add(
+            (new ObjectField(
+                'adyen_data',
+                'adyenData'
+            ))->addFlags(new Runtime(), new Computed(), new ApiAware())
```

Comment:
is the `ApiAware` flag really necessary? I see from it's definition that it only has the following whitelisted:
```
AdminApiSource::class => '/api/',
SalesChannelApiSource::class => '/sales-channel-api/',
```
now that we no longer use sales-channel-api won't this be a problem?

Reply:
@peterojo actually since 6.3.5.1 it is required for this data to be shown in the store api. not sure where exactly they doe the mapping but it actually supports store api without any additional stuff.

However, if you want to support older SW versions then 6.3.5.1,  I will have to remove it but then it might be lost in 6.3.5 as they switched from blacklisting to whitelisting in 6.3.5.1

I could check if the class exists and only add the flag if it exists, then it should support new and old versions

Reply:
I now check for class existince to ensure it works with previous versions too. However, this flag is enough for the store api without any adjustments. tested in multiple projects now. Not sure why the base url for the store api is not specified in the flag though but it does work and this is how shopware does it internally now too in all their definitions.

Reply:
Great, we would prefer to still continue to support all versions from v6.3.3.1 and above for as long as possible

Reply:
@peterojo Ok. i retested and without the ApiAware flag and 6.3.5.1 the data is missing. with it is shown. and the class check should ensure it still works fine for older version.